src/Revenue/Migrations/AddPastDonationsToRevenueTable.php

```diff
@@ -114,4 +103,33 @@ public static function id() {
 	public static function timestamp() {
 		return strtotime( '2019-09-24' );
 	}
+
+	/**
+	 * Pause update process and add log.
+	 *
+	 * @param Give_Updates $give_updates
+	 *
+	 * @since 2.9.2
+	 */
+	private function pauseUpdateOnError( $give_updates ) {
```

Comment:
@JasonTheAdams This code can be reused in every data update. I think we can refactor.

What do you think?

Reply:
@ravinderk This is great! Can you explain what pausing means? Does it mean that the user will be able to re-run that specific batch when they resume?

Please make an issue to add this functionality, I don't want to modify too much in this PR as it's a bug release. I'd like to talk through migrations more when I put together the pitch.

Reply:
> @ravinderk This is great! Can you explain what pausing means? Does it mean that the user will be able to re-run that specific batch when they resume?

@JasonTheAdams When I call the pause function it will stop update process at specific point and store state of update. When admin restart update then updates will resume from the past state.

> Please make an issue to add this functionality, I don't want to modify too much in this PR as it's a bug release. I'd like to talk through migrations more when I put together the pitch.

Agree

Reply:
Awesome! Thanks for putting that together!